lib/src/components/NxToggle/NxToggle.scss

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+@import '../../scss-shared/nx-variables';
+@import '../../scss-shared/nx-text-helpers';
+
+.nx-toggle {
+  display: flex;
+  margin-bottom: $nx-spacing-md;
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  &.nx-toggle--disabled {
+    &, &:hover {
+      color: $nx-text-color-disabled;
+      cursor: default;
+
+      .nx-toggle__control {
+        background-color: $nx-grey-200;
+        border-color: $nx-grey-500;
+      }
+      .nx-toggle__indicator {
+        background-color: $nx-grey-500;
+      }
+    }
+  }
+
+  &:focus-within {
+    .nx-toggle__control {
+      border-color: $nx-focus-border-color;
+      filter: drop-shadow(0 0 3px rgba($nx-focus-border-color, 0.4));
+
+      &:hover {
+        border-color: $nx-grey-1100;
+      }
+    }
+  }
+}
+
+.nx-toggle__input {
+  margin: 0;
+  position: absolute;
+  width: 0;
+  z-index: -1;
+
+  &:checked {
+    + .nx-toggle__control {
+      background-color: $nx-blue-600;
+
+      .nx-toggle__indicator {
+        background-color: #fff;
+        left: 23px;
+      }
+    }
+  }
+}
+
+.nx-toggle__content {
+  flex-grow: 1;
+}
+
+.nx-toggle__control {
+  background-color: #fff;
+  border: 1px solid $nx-blue-600;
+  border-radius: 12px;
+  box-sizing: border-box;
+  height: 24px;
+  margin-left: $nx-spacing-l;
+  padding: $nx-spacing-xxs - 1px;
+  position: relative;
+  min-width: 44px;
+  max-width: 44px;
```

Comment:
Is there a reason this is specified using an identical max and min and not simply `width`?

Reply:
> Is there a reason this is specified using an identical max and min and not simply `width`?

Using just `width` it would squash when the label text was long. I could probably get around that by setting `flex-basis` on the container but at this point I think that's probably 6 of 1, half a dozen of the other.

Reply:
> Using just width it would squash when the label text was long. I could probably get around that by setting flex-basis on the container but at this point I think that's probably 6 of 1, half a dozen of the other.

I think the issue is actually `flex-shrink` defaulting to 1.  The cleanest solution IMO would be to set `flex: none` (and `width`).

Reply:
> > Using just width it would squash when the label text was long. I could probably get around that by setting flex-basis on the container but at this point I think that's probably 6 of 1, half a dozen of the other.
> 
> I think the issue is actually `flex-shrink` defaulting to 1. The cleanest solution IMO would be to set `flex: none` (and `width`).

Ok: https://github.com/sonatype/sonatype-react-shared-components/pull/250/commits/d2637515314ec2687da62ae81342fec992203ceb